File: javascript-leetcode/50.01.js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function(x, n) {
  if (n < 0) {
    x = 1 / x;
    n = -n;
  }
  let pow = 1;
  while (n > 0) {
    if ((n & 1) === 1) {
      pow *= x;
    }
    x *= x;
    n >>>= 1;
  }
  return pow;
};

const result = myPow(2.0, -2147483648);
console.log(result);

File: python-leetcode/50.01.py
class Solution:
  def myPow(self, x: float, n: int) -> float:
    if n < 0:
      x = 1 / x
      n = -n
    result = 1
    while n > 0:
      if (n & 1) == 1:
        result *= x
      x *= x
      n >>= 1
    return result
